client/core/trade.go

```diff
@@ -527,6 +527,50 @@ func (t *trackedTrade) counterPartyConfirms(match *matchTracker) (have, needed u
 	return
 }
 
+// deleteStaleCancelOrder checks if this trade has an associated cancel order,
+// and deletes the cancel order if it (i.e. the cancel order) stays at Epoch
+// status for 2 or more epochs after the close of the cancel order's epoch.
+// Deleting the stale cancel order from this trade makes it possible for the
+// client to re-attempt cancelling the order.
+//
+// NOTE:
+// Stale cancel orders would be Executed if their preimage was sent or Revoked
+// if their preimages was not sent. We cannot currently tell whether the cancel
+// order's preimage was revealed, so assume that the cancel order is Executed
+// but unmatched. Consider adding a order.PreimageRevealed field to ensure that
+// the correct final status is set for the cancel order.
+//
+// This method MUST be called with the trackedTrade mutex lock held for writes.
+func (t *trackedTrade) deleteStaleCancelOrder() {
+	if t.cancel == nil || t.metaData.Status != order.OrderStatusBooked {
+		log.Debugf("Ignoring delete cancel order attempt for order %v, status %v, pending cancel = %v",
+			t.ID(), t.metaData.Status, t.cancel != nil)
+		return
+	}
+
+	stamp := t.cancel.ServerTime
+	epoch := order.EpochID{Idx: encode.UnixMilliU(stamp) / t.epochLen, Dur: t.epochLen}
+	epochEnd := epoch.End()
+	if time.Since(epochEnd).Milliseconds() < int64(2*t.epochLen) {
+		return // not stuck, yet
+	}
+
+	log.Infof("Cancel order %v in epoch status with server time stamp %v, epoch end %v (%v ago) considered executed and unmatched.",
+		t.cancel.ID(), t.cancel.ServerTime, epochEnd, time.Since(epochEnd))
+
+	err := t.db.LinkOrder(t.ID(), order.OrderID{})
+	if err != nil {
+		log.Errorf("DB error unlinking cancel order %s for trade %s: %w", t.cancel.ID(), t.ID(), err)
+	}
+	// Clearing the trackedCancel allows this order to be canceled again.
+	t.cancel = nil
```

Comment:
What if the dexConnection is just down and the trade has been canceled?  Can that happen?  If it does, would the actually-cancelled trade just be reconciled on connect?

Reply:
My reply to your previous comment answers this somewhat. We currently only clear stale cancel orders **after** the server reports the targeted orders as active (Epoch/Booked status) in the connect response.